### PR TITLE
Notification

### PR DESCRIPTION
**Pull request by dhanzy**
### What this PR does (required):
- Number of Notification received shows on navbar

### Screenshots / Videos (required):
- Screenshot (32)
- 
Screenshot (37)

### Any information needed to test this feature (required):
- Sign in
- Click on the notification panel on the navbar

### Any issues with the current functionality (optional):
- Does not display notifications when hovered yet

### CODE REVIEW

File: client/src/components/Navbar/Navbar.tsx [hunk 1]
```diff
@@ -18,6 +21,8 @@ const Navbar = (): JSX.Element => {
     setAnchorEl(event.currentTarget);
   };
 
+  console.log('Notificaltions');
```
Comment (rajivtitus) [inline client/src/components/Navbar/Navbar.tsx]: Please remove both the console.log's

File: client/src/pages/Notifications/Notification.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import { Grid, Typography, Box } from '@material-ui/core';
+import { useNotifications } from '../../context/useNotificationContext';
+import SidePanel from '../../components/SidePanel/SidePanel';
+
+function Notification(): JSX.Element {
+  const { allNotifications, readNotification } = useNotifications();
+  console.log(allNotifications);
```
Comment (rajivtitus) [inline client/src/pages/Notifications/Notification.tsx]: Please remove

File: server/controllers/notification.js [hunk 3]
```diff
@@ -1,20 +1,26 @@
...
             sender,
             content,
             notificationType,
-            receiver
+            receiver: receiver._id,
         }).save();
         if (!notification) {
             return res.status(400).json({ status: "submission not saved"})
         }
+        console.log('Saved');
```
Comment (rajivtitus) [inline server/controllers/notification.js]: Please remove

File: client/src/pages/SubmitPage/Submit.tsx [hunk 4]
```diff
@@ -80,6 +83,13 @@ function FileUploader(): JSX.Element {
                 updateSnackBarMessage(
                   `Submitted ${imageList.length} images to ${contest == undefined ? 'Contest' : contest.title}`,
                 );
+                {
+                  contest.userID == undefined
+                    ? updateSnackBarMessage('Could not send Notification to contest owner')
+                    : createSubmitNotification({ receiverID: contest.userID._id }).then((resp) => {
+                        console.log(resp);
```
Comment (rajivtitus) [inline client/src/pages/SubmitPage/Submit.tsx]: Please remove

File: client/src/pages/SubmitPage/Submit.tsx [hunk 5]
```diff
@@ -80,6 +83,13 @@ function FileUploader(): JSX.Element {
                 updateSnackBarMessage(
                   `Submitted ${imageList.length} images to ${contest == undefined ? 'Contest' : contest.title}`,
                 );
+                {
+                  contest.userID == undefined
```
Comment (rajivtitus) [inline client/src/pages/SubmitPage/Submit.tsx:112]: What is the reason for the ternary operator here? If contest.userID is undefined won't we always be return a loading spinner?

**Review by rajivtitus — COMMENTED** [review]
LGTM! Just some minor cleanup required, please have a look at comments and address. Thanks!

File: client/src/pages/SubmitPage/Submit.tsx (hunk 5 above)
Comment (dhanzy) [reply client/src/pages/SubmitPage/Submit.tsx:112]: Yes I defined contest.userID == undefined at the top to return a loading spinner but for some odd reasons typescript still complains contest.userID is undefined

Comment (rajivtitus) [reply client/src/pages/SubmitPage/Submit.tsx:112]: Maybe check your interfaces as well as how your passing a contest into this component?

**Review by rajivtitus — APPROVED** [review]
Please remove all console.log's (I think you've missed a few) before merging. Thank you!

**Review by HAZberi — APPROVED** [review]
Please perform the required clean up and merge. Thank you.

**Pull request merged**